packages/lib/gpc/src/gpcCompile.ts

```diff
@@ -360,42 +362,64 @@ function compileProofListMembership<
   paramMaxListElements: number
 ): {
   listComparisonValueIndex: CircuitSignal[];
+  listContainsComparisonValue: CircuitSignal[];
   listValidValues: CircuitSignal[][];
 } {
   // Arrange list names alphabetically.
   const listNameOrder = Object.keys(listConfig).sort();
 
-  // Do the same for the lists of comparison IDs and string them together.
+  // Arrange for membership before non-membership checks and consider the
+  // lists of comparison IDs in alphabetical order, stringing them all together.
   // This takes into account the possibility where different entry (or tuple)
   // values must lie in the same list, or the case one entry (or tuple) value
   // must lie in multiple lists.
-  const listIdPairs = listNameOrder
-    .map((listName: PODName): [PODName, PODEntryIdentifier][] =>
-      listConfig[listName].sort().map((elementId) => [listName, elementId])
+  const listIndicatorIdTriples = listNameOrder.flatMap((listName) =>
+    (Object.keys(listConfig[listName]).sort() as ListMembershipEnum[]).flatMap(
+      (membershipIndicator) =>
+        listConfig[listName][membershipIndicator]
+          .sort()
+          .flatMap((elementId) => [
+            [listName, membershipIndicator, elementId] as [
```

Comment:
Why is the `as` necessary here?  Seeing a type is useful to me as a reader, but I know that `as` can bypass type-checking, which is worrying.  Would `satisfies` work instead?  Or if you declared the return-type of the lambda, or of the array at the top level would it make the type here unnecessary?